src/appsettings/validators.py

```diff
@@ -67,3 +69,43 @@ def __call__(self, value):
             if not isinstance(element, self.value_type):
                 params = {"key": key, "value": element, "type": self.value_type.__name__}
                 raise ValidationError(self.message, params=params)
+
+
+class FileValidator(object):
+    """Validator which checks file existence and open mode permissions."""
+
+    not_exist_message = "File %(value)s does not exist."
+    no_perms_message = "You have no permission to open %(value)s with mode %(mode)s."
+    unexpected_error_message = "Unexpected error occurred while opening the file %(value)s: %(exception)s"
+
+    def __init__(
+        self, exists=True, mode="r", not_exist_message=None, no_perms_message=None, unexpected_error_message=None
+    ):
+        """
+        Try to open a file using a given ``mode``.
+
+        Args:
+            exists (bool): Require existence of the file
+            mode (str): ``mode`` argument for ``open`` builtin function.
+            not_exist_message (str): Error message (containing ``value``)
+            no_perms_message (str): Error message (containing ``value`` and ``mode``)
+            unexpected_error_message (str): Error message (containing ``value`` and ``exception``)
+        """
+        self.exists = exists
+        self.mode = mode
+        self.not_exist_message = not_exist_message or self.not_exist_message
+        self.no_perms_message = no_perms_message or self.no_perms_message
+        self.unexpected_error_message = unexpected_error_message or self.unexpected_error_message
+
+    def __call__(self, value):
+        try:
+            with open(value, mode=self.mode):
```

Comment:
It's not possible to check if the path is a directory.

Reply:
Well, I don't think it should be a directory... For a directory we could create a new `Setting`.

If you do want to make both files and directories setting, I would suggest to create the previous validation based on `os.access()`. Another reason to do it might be [my comment below](https://github.com/pawamoy/django-appsettings/pull/81/#pullrequestreview-386395778). But still - I would rather have two individual subsetting classes.

Reply:
It seems they would be rather duplicated. I don't really see what would be the difference between setting for a file and a directory.

Reply:
When I see `FileSetting`, I expect it to be a file, not a directory. When I see `DirectorySetting`, I expect it to be a directory.

But OK, not it should be able to take a dir as well.

Reply:
Well, technically directory is also a file :)